sql_assesment.txt

```diff
@@ -0,0 +1,1786 @@
+
++------+--------+---------+------+
+| SNUM | SNAME  | CITY    | COMM |
++------+--------+---------+------+
+| 1002 | Serres | SanJose |   13 |
+| 1004 | Motika | London  |   11 |
+| 1008 | Fran   | London  |   25 |
++------+--------+---------+------+
+
+
+
+
+67) Write a query that selects all customers serviced by Peel or Motika. (Hint: The snum field relates the 2 tables to one another.) 
+
+
+select * from customers where snum in (select snum from salespeople where sname= "Peel" or sname= "Motika");
+
++------+---------+--------+--------+------+
+| CNUM | CNAME   | CITY   | RATING | SNUM |
++------+---------+--------+--------+------+
+| 2001 | Hoffman | London |    100 | 1001 |
+| 2006 | Ciemens | London |    100 | 1001 |
+| 2007 | Periera | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+
+
+
+
+68) Count the number of salespeople registering orders for each day. (If a salesperson has more than one order on a given day, he or she should be counted only once.)
+
+select odate,count(distinct(snum)) from orders group by odate;
+
++------------+-----------------------+
+| odate      | count(distinct(snum)) |
++------------+-----------------------+
+| 1990-03-10 |                     4 |
+| 1990-04-10 |                     2 |
+| 1990-05-10 |                     1 |
+| 1990-06-10 |                     2 |
++------------+-----------------------+
+
+
+
+
+69) Find all orders attributed to salespeople who live in London. 
+
+
+select * from orders where snum in (select snum from salespeople where city="London");
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3003 |  767.19 | 1990-03-10 | 2001 | 1001 |
+| 3002 |  1900.1 | 1990-03-10 | 2007 | 1004 |
+| 3008 |    4723 | 1990-05-10 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-06-10 | 2006 | 1001 |
++------+---------+------------+------+------+
+
+
+
+
+
+70) Find all orders by customers not located in the same cities as their salespeople. 
+
+
+select * from orders where cnum in (select cnum from customers where snum in (select snum from salespeople where customers.city!= city));
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-03-10 | 2008 | 1007 |
+| 3002 |  1900.1 | 1990-03-10 | 2007 | 1004 |
+| 3006 | 1098.16 | 1990-03-10 | 2008 | 1007 |
+| 3009 | 1713.23 | 1990-04-10 | 2002 | 1003 |
+| 3007 |   75.75 | 1990-04-10 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-06-10 | 2004 | 1002 |
++------+---------+------------+------+------+
+
+
+
+
+71) Find all salespeople who have customers with more than one current order. 
+
+
+select s.*, sname , count(onum) from salespeople s, customers c, orders o where s.snum= c.snum and o.cnum= c.cnum group by snum having (count(onum)>1);
+
```

Comment:
Incorrect. GROUP BY must be used on cnum. Please correct it.
And avoid using * as much as possible.